json/json.go

```diff
@@ -0,0 +1,115 @@
+// Package json provides a type and helpers for representing JSON log
+// lines.
+//
+// Format
+//
+// The JSON log line format consists of JSON objects, one per
+// \n-terminated line.
+//
+// Unilog recognizes three fields in that object that are considered
+// special (all are optional):
+//
+//    - timestamp: The time stamp of an event. Unilog understands both
+//      epoch timestamps as float, or RFC3339Nano-formatted
+//      timestamps. The timestamp will be normalized to
+//      nanosecond-resolution float "timestamp" fields.
+//    - canonical: Identifies the log event as "canonical", i.e. the
+//      most important line a service can log. It is considered to have
+//      the highest criticality level.
+//    - clevel: The criticality level of the event.
+//
+// Example
+//
+//    {"timestamp":"2006-01-02T15:04:05.999Z07:00","message":"hi there"}
+//    {"timestamp":"2006-01-02T15:04:05.999Z07:00","message":"hi there"}
+package json
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"time"
+)
+
+// JSONLogLine is a representation of a generic log line that unilog
+// can destructure.
+type LogLine map[string]interface{}
+
+const timestampField = "timestamp"
+
+var tsFields = []string{
+	timestampField,
+	"ts",
+}
+
+// Timestamp returns the timestamp of a log line; if a timestamp is
+// set on the line, Timestamp will attempt to interpret it
```

Comment:
Do we really need to parse the timestamp value? I don't expect we have control over this in every log, and it's easier to tell Splunk how to parse it than make every input log file the same (or add support for every potential input format into this binary)